distributed/scheduler.py

```diff
@@ -1635,6 +1637,37 @@ def get_dataset(self, stream, name=None, client=None):
         else:
             raise KeyError("Dataset '%s' not found" % name)
 
+    @gen.coroutine
+    def register_environments(self, stream=None, environments=None, workers=None):
+        if environments is None:
+            environments = self.environments
+        else:
+            for name in environments:
+                if name in self.environments:
+                    raise KeyError("Environment %s already exists" % name)
+
+        if workers is None:
+            workers = list(self.worker_info)
+
+        results = yield [self.rpc(addr=worker).register_environments(environments=environments)
+                         for worker in workers]
+
+        exceptions = {}
+        for worker, result in zip(workers, results):
+            for name, val in result.items():
+                if isinstance(val, bool):
+                    if name not in self.environments:
+                        self.environments[name] = environments[name]
+                        self.environment_workers[name] = set()
+                    if val:
+                        self.environment_workers[name].add(worker)
+                else:
+                    if name not in exceptions:
+                        exceptions[name] = [val]
+                    else:
+                        exceptions[name].append(val)
+        raise Return(exceptions)
```

Comment:
That's doable. Since these can change as workers are added, perhaps making `environment_workers` accessible on the client might be nice as well (would probably call it `environments` and make it a property that returns a dictionary).